scanpy/plotting/_tools/__init__.py

```diff
@@ -633,3 +636,110 @@ def sim(adata, tmax_realization=None, as_heatmap=False, shuffle=False,
                    highlightsX=np.arange(tmax, n_realizations*tmax, tmax),
                    xlabel='index (arbitrary order)')
         utils.savefig_or_show('sim_shuffled', save=save, show=show)
+
+
+def embedding_density(
+        adata: AnnData,
+        basis: str,
+        key: str,
+        group: Optional[str] = None,
+        color_map: Union[Colormap, str] = 'YlOrRd',
+        greysize: Optional[int] = 20,
+        colorsize:  Optional[int] = 40,
+        vmax:  Optional[int] = 1,
+        vmin:  Optional[int] = 0,
+        save: Union[bool, str, None] = None,
+        **kwargs):
+    """Plot the density of cells in an embedding (per condition)
+
+    Plots the gaussian kernel density estimates (over condition) from the
+    `sc.tl.density()` output.
+
+    This function was written by Sophie Tritschler and implemented into
+    Scanpy by Malte Luecken.
+    
+    Parameters
+    ----------
+    adata : :class:`~anndata.AnnData`
+        The annotated data matrix.
+    basis : `str`
+        The embedding over which the density was calculated. This embedded
+        representation should be found in `adata.obsm['X_[basis]']``.
+    key : `str`
+        Name of the `.obs` covariate that contains the density estimates
+    group : `str`, optional (default: `None`)
+        Category of the observed covariate which will be plotted.
```

Comment:
is not clear to me from the description what group means or does. Is a categorical value from `.obs` ? If this is the case, in other tools we tend to use `groupby`.

Reply:
It's a specific category from a categorical covariate in `.obs`. I struggle to define this in a nicer way.

Reply:
maybe similar to sc.pl.umap (and other plotting functions)
groups : str, None
Restrict to a few categories in categorical observation annotation. The default is not to restrict to any groups.

Reply:
Good idea, thanks. I don't think I can use `plot_scatter` to plot multiple categories like this though, because you need different dot sizes, and density values for each category. I don't think I can just wrap all those things in a list for multiple plots.